starknet_py/utils/typed_data.py

```diff
@@ -20,6 +23,50 @@ class Parameter:
     type: str
 
 
+@dataclass
+class Domain:
+    """
+    Dataclass representing a domain object (StarkNetDomain, StarknetDomain)
+    """
+
+    name: str
+    version: str
+    chain_id: Union[str, int]
+    revision: Optional[Union[str, int]] = None
+
+    def __post_init__(self):
+        self.resolved_revision = (
+            Revision(self.revision) if self.revision else Revision.V0
+        )
+        self.separator_name = self._resolve_separator_name()
+
+    def _resolve_separator_name(self):
+        if self.resolved_revision is Revision.V0:
```

Comment:
Generally it is recommended to use `==`, is there any specific reason to use `is` here?
```suggestion
        if self.resolved_revision == Revision.V0:
```

Reply:
Yes, let's use `==` as it compares the actual value 👍